W3ChampionsStatisticService/Startup.cs

```diff
@@ -76,9 +79,23 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSpecialBsonRegistrations();
 
             services.AddSingleton<TrackingService>();
-            services.AddSingleton<PlayerAkaProvider>();
-            services.AddSingleton<PersonalSettingsProvider>();
-            services.AddSingleton<MatchmakingProvider>();
+            services.AddTransient<PlayerAkaProvider>();
+            services.AddTransient<PersonalSettingsProvider>();
+            services.AddTransient<MatchmakingProvider>();
+
+            services.AddMemoryCache();
+            services.AddTransient(typeof(ICacheData<>), typeof(InMemoryCacheData<>));
```

Comment:
Most developers make a wrong decision about services lifetime. 
Almost all services, controllers we have in a single representation, but everything should be Transient by default to support correct dependency injection. 
In fact singletons can only be cache, http clients, maybe db connection. Because they bound to server instance
Scoped only User Information, Or request cache between services.
Our service should be Transient.
services.AddMemoryCache(); 
this line add .NET memory cache as singleton, but it is hidden behind intentionally.